File: spring/CadenaSuministrosDB2/src/main/java/com/uvg/cadenasuministrosdb2/app/domain/Suppliers.java
package com.uvg.cadenasuministrosdb2.app.domain;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.neo4j.core.schema.GeneratedValue;
import org.springframework.data.neo4j.core.schema.Id;
import org.springframework.data.neo4j.core.schema.Node;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Node("Suppliers")
public class Suppliers {

    @Id
    @GeneratedValue
    private Long id;

    private Integer supplierId;
    private String name;
    private String address;
    private String phone;
    private String country;
    private Integer reputation;
    private String contact;
    private String email;
    private String website;
}

File: front/src/pages/SuppliersNode/Suppliers.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import DataTable from '../../components/DataTable';
import { Typography } from '@mui/material';

function Suppliers() {
  const [suppliers, setSuppliers] = useState([]);

  useEffect(() => {
    fetchSuppliers();
  }, []);

  // http://18.191.199.191:1913/api/inventories/56270
  const fetchSuppliers = async () => {
    try {
        const response = await axios.get('http://3.145.98.75:1913/api/suppliers');
        setSuppliers(response.data);
    } catch (error) {
        console.error('Error fetching data:', error);
    }
  };

  return (
      <div>
          <Typography variant="h4" gutterBottom>
              Proveedores
          </Typography>
          <DataTable
              table={{
                  columns: [
                      { Header: "Id", accessor: "id" },
                      { Header: "Name", accessor: "name" },
                      { Header: "Address", accessor: "address" },
                      { Header: "Phone", accessor: "phone" },
                      { Header: "Country", accessor: "country" },
                      { Header: "Reputation", accessor: "reputation" },
                      { Header: "Contact", accessor: "contact" },
                      { Header: "Email", accessor: "email" },
                      { Header: "Website", accessor: "website" },
                  ],
                  rows: suppliers
              }}
          />
      </div>
  );
}

export default Suppliers;
